docs/INTRO.md

```diff
@@ -0,0 +1,441 @@
+```
+
+Thus, any time we see the kind `S -> Type`, we should mentally substitute its kind synonym `PType`. We reiterate: types of kind `PType`, should be considered as _tags_ on computation. They do not represent types of values in the same way as standard Haskell types. 
+
+The kind of basic types such as `Integer` in Haskell is `*`; the corresponding "basic" kind in Plutarch is simply `PType`. Higher-kinded types in Haskell, such as `Maybe`, will kinds such as `* -> *`. In Plutarch, the corresponding kind is:
+
+```hs
+ghci> :k PMaybe
+PMaybe :: PType -> S -> Type
+```
+
+Since the kind arrow `->` is right-associative, we first read this as `PMaybe :: PType -> (S -> Type)`; and since we know that that `PType` and `S -> Type` and synonyms, we read this as `PMaybe :: PType -> PType`, which should agree without intuition.
+
+The kind `S -> Type` is mysterious at first, but we recall that `PType`s are _tags_ on (unexecuted) computations indicating their result type. The `S` kind represents the computational context; thus, a `PType` expects to receive a _computational context_ represented by a value `s` whose type has kind `S` that it will tag to produce a `Type`. Note that end-users never instantiate the value `s` with a concrete value; it is simply a type-level mechanism to maintain functional purity.
+
+The above notion is essential to understanding why not all `PType`s have data constructors; the data constructors are irrelevant, except insofar as they enable the implementation to keep track of Haskell-level and UPLC-level representations. `PInteger` is one such case; it is impossible to construct a constant `y` where `y = PInteger`. Other `PType`s, such as `PMaybe`, _do_ have data constructors (specifically `PJust` and `PNothing`), but _still_ do not carry data from the viewpoint of UPLC. A value such as `PNothing` indicates that the Plutarch compiler should interpret the `Data` representation of the result of an executed UPLC script as `PNothing` (bridging UPLC with Plutarch). The script's result should type check and carry the same semantics as Haskell's `Nothing`; we imply nothing more and nothing less. 
+
+# Plutarch `Term`s
+
+Plutarch `Term`s are terms in the sense of simply-typed lambda calculus terms. In a lambda calculus, we can construct terms as either "constants" or "lambdas," and terms can either be "open" (having free variables) or "closed" (having no free variables). We compose Plutarch `Term`s to build up increasingly complex computations. Once all free variables are eliminated from a `Term` (making it a `Closed Term`), we can compile it using the eponymous function from the `Plutarch` module:
+
+```hs
+-- | Closed term is a type synonym 
+type ClosedTerm (a :: PType) = forall (s :: S). Term s a 
+
+-- | Compile operates on closed terms to produce usable UPLC scripts.
+compile :: ClosedTerm a -> Script
+```
+`Term`s are constructed from Haskell values and are tagged with `PType`s.
+
+## Plutarch Constant `Term`s
```

Comment:
I think the details of the typeclasses (`PConstant`/`PLift` and `PlutusType`/`PCon`/`PMatch`) should stay at their own section in the Typeclasses document. The constant building could stay surface level here, like it was before. What do you think?

Reply:
I do agree with that. I'm just a little concerned about too much duplicate info and the guide getting too big and unwieldy. But I think you'd actually be a much better judge for whether or not the guide, at its current stage, seems too big.

My concern is that there's *a lot* of really crucial and helpful information in all the corners of the guide - but when it gets too big, no one's going to read all of it and hence the information just sits there. I'm hopeful that the splitting effort will make it easier to grasp - and perhaps I'll scrape off some info from other sections if a lot of it will be covered in intro.

Reply:
Yeah, I think my (personal) ideal guide has a pretty big intro. My goal is that you read this big intro/overview once to get a picture of how everything is laid out, and then don't really need to go back to it ever again because you've got tutorials, how-tos, and reference documentation that is better suited for actual development.

Perhaps we should make this clear, and think about what goes where. I think redundancy and repetition can be good in some scenarios -- its annoying to have to hop around from resource to resource -- but each time the same information is stated should be A.) be consistent, and B.) serve a distinct, explicit purpose.

Perhaps we should consider just going full-bore on the diataxis framework and plan on separating out 

- Explain: call this "The Guide" and include `INTRO.md` and some sections from concepts/usage
- Reference: The typeclass/types docs, and maybe some additional function reference material that aren't fully suitable for haddock comments, plus some of the shorter examples
- Tutorials: when we get there, building an end-to-end app using plutarch, or showing real-world cases of optimization, common mistakes, etc. Long-form examples.
- HOWTOs: fold in the tips & tricks, common issues and trouble shooting, short-form examples, etc.

I think that would be out of scope for this PR, but useful going forward. Thoughts?

Reply:
That sounds good to me. I'll keep that in mind. But I don't think I'll be able to do it during this split. Perhaps you could make an issue about this as well that we can track. Otherwise, I will put it in my todo list.

Reply:
Done, spun it off. into a new issue. Happy to contribute as time allows -- or as my own needs necessitate it :smile: